packages/block-editor/src/components/block-toolbar/index.native.js

```diff
@@ -1,105 +1,41 @@
 /**
- * External dependencies
+ * WordPress dependencies
  */
-import { View, ScrollView, Keyboard, Platform } from 'react-native';
+import { withSelect } from '@wordpress/data';
 
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
-import { Toolbar, ToolbarButton, Dashicon } from '@wordpress/components';
 import { BlockFormatControls, BlockControls } from '@wordpress/block-editor';
-import { __ } from '@wordpress/i18n';
-
-/**
- * Internal dependencies
- */
-import styles from './style.scss';
-
-export class BlockToolbar extends Component {
-	constructor() {
-		super( ...arguments );
 
-		this.onKeyboardHide = this.onKeyboardHide.bind( this );
+export const BlockToolbar = ( { blockClientIds, isValid, mode } ) => {
```

Comment:
Not a big deal at all, but what would you think about combining the render-nothing cases instead of having two different ones (the first returning `null` and the second returning `<></>`? To my eyes, something like this is easier to follow:

```
export const BlockToolbar = ( { blockClientIds, isValid, mode } ) => {
	const showToolbar = blockClientIds.length !== 0 && mode === 'visual' && isValid;
	return showToolbar ? (
			<>
				<BlockControls.Slot />
				<BlockFormatControls.Slot />
			</>
		) : null; 
};
```

I don't feel strongly about this, so feel free to ignore if you prefer it as-is.

Reply:
Indeed it's a bit better 👍
I think we can keep the current syntax though as it's closer to the web version and we don't delay this PR much longer as well. Let's revisit later if necessary